deployer/src/runtime_manager.rs

```diff
@@ -115,19 +131,56 @@ impl RuntimeManager {
         .await
         .context("failed to start shuttle runtime")?;
 
+        let process = Arc::new(Mutex::new(process));
+        let process_cloned = Arc::clone(&process);
         self.runtimes
             .lock()
             .unwrap()
             .insert(id, (process, runtime_client.clone()));
 
+        let stdout = process_cloned
+            .lock()
+            .await
+            .stdout
+            .take()
+            .context("child process did not have a handle to stdout")?;
+
+        // Ensure the child process is spawned in the runtime so it can
+        // make progress on its own while we await for any output.
+        tokio::spawn(async move {
+            process_cloned
+                .lock()
+                .await
+                .wait()
+                .await
+                .expect("child process encountered an error");
+        });
+
+        let mut reader = BufReader::new(stdout).lines();
+        let mut logger_client = self.logger_client.clone();
+        tokio::spawn(async move {
```

Comment:
I am wondering if it makes sense to handle this logging related task in the outer scope. Probably it means propagating this up to `task` from `run.rs`.

Reply:
`run.rs` might not have access to the logger client though

Reply:
I meant just propagating the task handle, no need for the client in that scope. In some cases logging sending might break so detecting clearly when that happens should help.